RevenueCatUI/PaywallView.swift

```diff
@@ -106,9 +138,18 @@ public struct PaywallView: View {
     ) {
         self._introEligibility = .init(wrappedValue: introEligibility ?? .default())
         self._purchaseHandler = .init(wrappedValue: purchaseHandler)
-        self._offering = .init(
-            initialValue: offering ?? Self.loadCachedCurrentOfferingIfPossible()
-        )
+        self._offeringSelection = .init(wrappedValue: offeringSelection)
+        var initialOffering: Offering?
+        if let offering {
+            initialOffering = offering
+        } else if let offeringSelection,
+                  Purchases.isConfigured,
+                  let cachedOfferings = Purchases.shared.cachedOfferings {
+            initialOffering = offeringSelection(cachedOfferings)
```

Comment:
it likely works the same, but still this is probably better:
```
        } else if let offeringSelection {
              if Purchases.isConfigured, let cachedOfferings = Purchases.shared.cachedOfferings {
                   initialOffering = offeringSelection(cachedOfferings)
              }
        } else {
            initialOffering = Self.loadCachedCurrentOfferingIfPossible()
        }
```

Reply:
Updated